lib/Model/Binary.cpp

```diff
@@ -156,6 +156,16 @@ bool Binary::verify(VerifyHelper &VH) const {
       return VH.fail();
   }
 
+  // Make sure no segments overlap
+  if (!Segments().empty()) {
+    for (const auto &[LHS, RHS] :
+         llvm::zip(skip_back(Segments()), skip_front(Segments()))) {
+      revng_assert(LHS.StartAddress() <= RHS.StartAddress());
+      if (LHS.endAddress() > RHS.StartAddress())
+        return VH.fail("Segments must not overlap.");
```

Comment:
Pass to fail at least one of the two, so it will get printed

Reply:
I went with a bit more explicit error message, but I can't imagine that's controversial. So I'm closing this.